lib/matplotlib/tests/test_datetime.py

```diff
@@ -260,11 +260,37 @@ def test_errorbar(self):
                      uplims=True, xuplims=True,
                      label='Data')
 
-    @pytest.mark.xfail(reason="Test for eventplot not written yet")
     @mpl.style.context("default")
     def test_eventplot(self):
-        fig, ax = plt.subplots()
-        ax.eventplot(...)
+        mpl.rcParams["date.converter"] = "concise"
+
+        fig, (ax1, ax2) = plt.subplots(2, 1, layout="constrained")
+
+        x_dates1 = np.array([datetime.datetime(2020, 6, 30),
+                             datetime.datetime(2020, 7, 22),
+                             datetime.datetime(2020, 8, 3),
+                             datetime.datetime(2020, 9, 14),],
+                            dtype=np.datetime64,
+                            )
+
+        ax1.eventplot(x_dates1)
+
+        start_date = datetime.datetime(2020, 7, 1)
+        end_date = datetime.datetime(2020, 10, 15)
+        date_range = end_date - start_date
+
+        dates1 = start_date + np.random.rand(30) * date_range
```

Comment:
If you want to use randomness, then you will need to set a seed (we use 19680801 for tests), or rewrite without randomness.

Reply:
Fixed both!